modules/javafx.controls/src/main/java/javafx/scene/control/skin/ProgressIndicatorSkin.java

```diff
@@ -742,7 +742,10 @@ private void rebuildTimeline() {
                 }
 
                 ((Timeline)indeterminateTransition).getKeyFrames().setAll(keyFrames);
-                indeterminateTransition.playFromStart();
+
+                if(NodeHelper.isTreeShowing(control)) {
+                    indeterminateTransition.playFromStart();
+                }
```

Comment:
In the case where `isTreeShowing` is false, and later goes to true, there will be a slight change in behavior, since `updateAnimation` will call `play` in that case. To make sure that the behavior is the same, I recommend calling `indeterminateTransition.jumpTo(Duration.ZERO);` in an `else` block here.